File: summerfield/ch01/hello/hello.py
# hello.py: signals & slots
import sys
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *
# using qdarkstyle (@see: https://github.com/ColinDuquesnoy/QDarkStyleSheet)
import qdarkstyle
# to detect dark themes (@see: https://pypi.org/project/darkdetect/)
import darkdetect

MIN_C, MAX_C = -100, 100

app = QApplication(sys.argv)
app.setStyle('Fusion')
# font = QFont("SF UI Text", 11)
app.setFont(QApplication.font("QMenu"))
if darkdetect.isDark():
    # apply dark stylesheet
    app.setStyleSheet(qdarkstyle.load_stylesheet(qt_api='pyqt5'))
print(f"Available styles {QStyleFactory.keys()}")
widget = QWidget()
widget.setWindowTitle("Hello")
layout = QVBoxLayout()
label = QLabel(f"Welcome to PyQt {PYQT_VERSION_STR} GUI programming!")
layout.addWidget(label)
widget.setLayout(layout)
widget.show()

sys.exit(app.exec_())

File: summerfield/ch01/hello/hello.cc
// hello.cc: Hello World with Qt
#include <QApplication>
#include <QHBoxLayout>
#include <QLabel>
#include <QPushButton>
#include <QVBoxLayout>

int main(int argc, char **argv)
{
    QApplication app(argc, argv);
    QApplication::setFont(QApplication::font("QMenu"));
    app.setStyle("Fusion");

    // create & show our GUI
    QWidget window;
    window.setWindowTitle("Hello Qt World");
    QVBoxLayout *l = new QVBoxLayout();
    QHBoxLayout *w = new QHBoxLayout;
    QLabel *hello = new QLabel("Welcome to Qt programming. Enjoy the ride!");
    QPushButton *quit = new QPushButton("Quit!");
    quit->setToolTip("Quit Application");
    w->addWidget(hello);
    w->addWidget(quit);
    l->addLayout(w);
    window.setLayout(l);

    // connect signals & slots
    QObject::connect(quit, SIGNAL(clicked()), qApp, SLOT(quit()));

    window.show();

    return app.exec();
}
